README.md

```diff
@@ -1,26 +1,49 @@
-<!--
-Avoid using this README file for information that is maintained or published elsewhere, e.g.:
+# namecheap-lego-operator
 
-* metadata.yaml > published on Charmhub
-* documentation > published on (or linked to from) Charmhub
-* detailed contribution guide > documentation or CONTRIBUTING.md
+## Description
 
-Use links instead.
--->
+Namecheap LEGO operator implements the provider side of the `tls-certificates-interface`
+to provide signed certificates from an ACME servers, using LEGO
+(https://go-acme.github.io/lego).
+It extends the `AcmeClient` charm from `acme_client` library to support Namecheap DNS API.
 
-# namecheap-lego-operator
+## Usage
+
+Deploy `namecheap-lego-operator`:
+
+If you wish to change the default configuration, create a YAML configuration file with fields you would like to change:
+
+`email` and `domain` are required fields.
+
+```yaml
+namecheap-lego:
+  email: <Account email address>
+  domain: <Domain to request certificate for>
+  namecheap-username: <Namecheap API user>
+  namecheap-api-key: <Namecheap API key>
+  namecheap-http-timeout: <API request timeout in seconds>
+  namecheap-polling-interval: <Time between DNS propagation checks in seconds>
```

Comment:
We can just  place the parameters that are necessary in the READMe example here (email, domain, username, api key) and add a `config` section where we detail all the config options including the optional ones.

Reply:
Agree, and also, to do anything, `namecheap-username` and `namecheap-api-key` are also required.

Reply:
✅